### ISSUE TITLE
Search for entry from the browser extension

### ISSUE THREAD

**Issue by CerebralFreeze** [keepassxreboot/keepassxc#5619]
## Summary
There are 2 reasons why I wrote this feature request.  First one is an important new upgrade, in my opinion.  The second one is not but simply triggers my OCD.

First, currently how KeePassXC is basically set up is provide only passwords for browsers.  (You may disagree with my assessment but that's not important for now.)  The important point is I believe productivity would skyrocket if when someone triggers the opening of a KeePassXC window, it will _almost always_ open the KeePassXC window so it shows the completion of a search of only the main domain name of the browser window that you were just looking at.  I italicized "almost always" because there are perhaps 1 or 2 scenarios in which you don't want this to happen.  For example, if you _just_ did a search and you are going back, you wouldn't want the search field to change at all because you probably went to the browser, something wrong occurred (you entered the wrong security password, you were looking at the wrong entry, etc.), and you immediately went back to the KeePassXC window.  As a result, you don't want anything to change with the window.  Your mind will still remember the point you left it at so if it's at a different window, you will be mentally confused.

Second feature that I think that should be made is that the Username field is not indented when you select an entry.  I know that the Username field has a pull down menu now.  But, it's annoying to me that it's indented and doesn't line up at the left side with all the other fields.  I just want to pull it back.  It seems like there is 2 spaces there that are a mistake.  If the program won't let you pull it back, perhaps indent all the other fields so it's moved 2 spaces to the right.

## Examples
For example, say you are trying to log into www.microsoft.com and that is the URL you are at.  You then open KeePassXC because you want to look at all the "Microsoft" accounts you have (meaning it'll do a search for Microsoft).  Once KeePassXC opens, it would be perfect if you could just select the Microsoft account that you need.

Right now instead, you have to clear any searches that might be there already.  Then, you have to click the search window.  (You have to click the search field because sometimes Control+F doesn't work.  Then you have to type Microsoft.  This is completely annoying and takes so much wasted time.  This would be an awesome feature upgrade.

What would be even better is if you could select the "tag" that the URL will trigger because Gmail is the name of the accounts that I use for Gmail but the URL is actually https://mail.google.com/ which doesn't have the word Gmail in it.  I would like to name my accounts Gmail and so it would search for that when I have mail.google.com up as the main domain in my browser window.

## Context
This matters because it would save a lot of time AND a lot of mental effort (willpower) during the day.  It really is aggravating when you have to open the KeePassXC window and you have to go through this entire procedure to find the password you are looking for.

**Comment by droidmonkey** [comment]
Why aren't you using the browser extension?

We will not implement the proposed feature request because it is no even possible to implement. How do we know what site you are looking at in your browser? What if that is not the action you want to take? Why would merely opening the app initiate an automated search?

Your second point is a duplicate.

**Comment by CerebralFreeze** [comment]
Howdy, Mr. Droid,

Uh...maybe I'm misinterpreting the tenor of your reply but it sounds like somebody didn't take their happy pills today.  Please don't yell at me because I'm very sensitive. 😭

I _think_ I'm using the browser extension properly.  I hope I am because I'll feel like a real idiot if I'm not after all this time.  Is there a way to use the browser extension that I don't know about?  How do you "use" the browser extension?  It's just installed with the settings set to work in a manner that you desire, no?  Is there a way to press it so it pops open a search or something?  The only way I "use" it is to left-click it (which I do a lot every day) because fields at frequently visited websites aren't still recognized.  So, I try to train the extension to distinguish between which fields are logins and passwords but for a lot of specific websites, it won't work sometimes.  Gmail is a big one that gives KeePassXC trouble.

Maybe it's just me but I have been using KeePassXC for about a year now but there are still _a lot_ of times every day in which a field is not recognized.  So, I have to call KeePassXC by tapping my designated global hotkey, "`".  Then, I can press Control+F to search for an entry but sometimes I can't.  (I don't know why it won't work even though the KeePassXC window is active.  I'll keep an eye on it from now on to figure out why Control+F won't work sometimes.)  But, even then, I still have to press Control+F and then think and type in the domain.  It would save so much time and mental energy if I am coming immediately from a browser, a search is automatically executed of the domain name of the URL.  I mean why else would you open up the KeePassXC window straight from a browser window, right?  I can honestly tell you that I do this _many_ times a day even though I've had KeePassXC for a while.  (It's pretty frustrating.)

**"How do we know what site you are looking at in your browser?"**

Hey, I'm not tech support!  Programming is your department, buddy.  My department is directly across the hall from you and I'm supposed to just submit complaints and unrealistic demands into your Inbox.  

But, if you really want a newbie to give it a go...  Don't yell at me if my answer sounds stupid since you're the one who asked. 
 How about this?  When the global hotkey for opening the KeePassXC is triggered, instead of immediately opening the window, have it see if the active window is a browser.  If it's a browser, have it check the URL box to see what the URL is.  Then, have it extract the domain from the URL by looking at what's after "www" or http or something like that.  Then, have KeePassXC put the domain into the search box and execute the search function so all possible entries are shown.

What's the global hotkey to open KeePassXC, you ask?  Well, I made it "`" for me.  I use a AutoHotkey script just for that.  Here it is:

```
`::
If (WinExist("ahk_exe KeePassXC.exe") && WinActive("ahk_exe KeePassXC.exe"))						
	WinMinimize
Else
{
	Run C:\Program Files\KeePassXC\KeePassXC.exe
	WinActivate, ahk_exe KeePassXC.exe
}
Return
```
The above script is way convenient since I toggle KeePassXC open/close that much.  Hey, I'll even let you add my script to KeePassXC for only $1 MM.  Just kidding.  😉  If you can, please totally add the above script/option to KeePassXC.  I totally think "`" should be used as a global toggle to open/close the KeePassXC window.  It's super-convenient.

Anyway, going back to the original issue, making KeePassXC window open up to search results when you are coming directly from a website (which is already stored in the KeePassXC database) would be a _huge_ time-saver for me.  It's probably by far the biggest daily time waster for me whenever I use KeePassXC.  Inputting new credentials is far more tedious and takes longer of course but since I've had KeePassXC for a while, I do that much less. 

**"What if that is not the action you want to take?"**

You can make it an option that people can select.  It's in the same nature like the option to clear out a search query after some allotted time.  It makes your work flow smoother.  That option is super-convenient too because after 20 minutes, you are not using the search query anymore and it's better if the KeePassXC window opens up to the main window.

**"Why would merely opening the app initiate an automated search? Too many possibilities for failure and annoyance to be of use. This is why the browser extension exists."**

I think that's a fair question.  But, if you think about it, why would you ever open a password manager app, coming directly from an active browser window?  I can think of only 3 possibilities.  First, you are coming from a new website to enter new credentials.  Two, you are coming from an old website that you already have credentials for but need to look it up.  Three, you are opening KeePassXC to adjust the settings.  There is no other reason.

For entering new credentials for a new website.  In that situation, if the domain doesn't lead to any search results, have it do nothing and open up to the main window.  Even in the very rare case it happens to unnecessarily show search results, who cares, no?

The second possibility is that you are at a URL that the password manager has the password to but something screwed up.  Or, perhaps you are looking up a secret security question.  Or, you are needing to do something else related to that website.  In these cases, it would be awesome if a search for the URL domain was completed and waiting for you to pick.  If you are already opened the entry and just going back and forth, the domain and the entry would match, no?  (Also, the search query timer wouldn't have elapsed.)  A search wouldn't be executed in this case.

The third possibility is you are adjusting settings.  Again, no harm, no foul.  It's just search results.

I don't think it would be an annoyance for users if you make it an option that people can pick just like clearing out the search query after 20 minutes.  This makes life so much easier for most users.  This option would actually be many times more useful than clearing out a search query which just requires just pressing Esc.  Both of these options are awesome because they are "mentally consistent".  It flows with your mental work flow.  It's what you want to see.

This would be the opposite of annoyance.  It would make life so much easier.  _Every_ password manager screws up a lot each and every day at websites that are already stored in the password database.  It would be nice if you are at the Gmail site and if you need to look up say a secret security question, you tap "`" and your Gmail accounts are already all waiting prettily on the main KeePassXC window, ready to be selected by you.

Can you pretty please consider it?  I think a lot of people would really like it.

> Why aren't you using the browser extension?
> 
> We will not implement the proposed feature request because it is not even possible to implement. How do we know what site you are looking at in your browser? What if that is not the action you want to take? Why would merely opening the app initiate an automated search? Too many possibilities for failure and annoyance to be of use. This is why the browser extension exists.
> 
> Your second point is a duplicate of #5601.

**Comment by droidmonkey** [comment]
I was asking my question in that accusatory, motherly kind of tone. So not angry, but you know you done something naughty.

How about this, because I did read your Steven King novel up there it got my black heart thinking a little. 

How about a button in the browser extension that calls keepassxc and performs a search for an entry. You could have it search for the current site domain (`url:google.com`) or a query of your choice. Then the "Allow sending entry to browser" window would show giving you the results. This would let you choose an entry and also optionally fix it up so it worked with the browser extension without the need for a search in the future.

**Comment by varjolintu** [comment]
@CerebralFreeze The next version of the extension improves a lot of sites, especially the ones with single username fields, including GMail. I hope it works as a healing oil for your sensitive skin 😁

**Comment by CerebralFreeze** [comment]
Psst...varjolintu.  Let me try to do the crying-child bit and let me see if I can get him to integrate it into settings as an automatic search option.  He's not really that bad like you told me.  Not sure why you said he's a brainless dodo who acts like a tyrant.  He seems pretty reasonable.  Maybe even a nice guy!

> @CerebralFreeze The next version of the extension improves a lot of sites, especially the ones with single username fields, including GMail. I hope it works as a healing oil for your sensitive skin 😁

**Comment by CerebralFreeze** [comment]
Mommy!  No, no, no, no!  [stomps his feet]  I **WANT** the KeePassXC to execute an automatic search for the domain/tag or **ELSE**.  And, I **WANT** it now!

Pretty please, Mommy Droid?  Please?  I promise I won't ask you for anything ever again.  Please?  😭

*In all seriousness, your last suggestion is probably best.  (I would _love_ integrating it into settings so that people have the option to automatically execute a search when coming directly from an active browser.)  I think your way is probably best because we can test to see if people will actually end up liking it.  I hope they do.  😊

With respect to doubling down, I don't know if you have read past issues I have submitted but I'm pretty OCD and need things to be perfect.  So, I was trying my darn hardest to convince you.  I can't help it.  😊

> I was asking my question in that accusatory, motherly kind of tone. So not angry, but you know you done something naughty.
> 
> How about this, because I did read your Steven King novel up there and it got my black heart thinking a little.
> 
> How about a button in the browser extension that calls keepassxc and performs a search for an entry. You could have it search for the current site domain (`url:google.com`) or a query of your choice. Then the "Allow sending entry to browser" window would show giving you the results. This would let you choose an entry and also optionally fix it up so it worked with the browser extension without the need for a search in the future.
> 


> BTW: it may not seem like it, but I appreciate you not crying in the corner when your idea is rejected. You double down and go for the personal touch. 😁

**Comment by droidmonkey** [comment]
Never wake a sleeping baby!

**Comment by droidmonkey** [comment]
This was implemented in 2.7.0, but just the protocol with an option to allow it.
